conans/test/functional/command/profile_test.py

```diff
@@ -14,140 +13,152 @@
-
-    @pytest.mark.tool("gcc")
-    @pytest.mark.skipif(platform.system() != "Darwin", reason="only OSX test")
-    def test_detect_default_in_mac_os_using_gcc_as_default(self):
-        """
-        Test if gcc in Mac OS X is using apple-clang as frontend
-        """
-        # See: https://github.com/conan-io/conan/issues/2231
-        output = check_output_runner("gcc --version", stderr=subprocess.STDOUT)
-
-        if "clang" not in output:
-            # Not test scenario gcc should display clang in output
-            # see: https://stackoverflow.com/questions/19535422/os-x-10-9-gcc-links-to-clang
-            raise Exception("Apple gcc doesn't point to clang with gcc frontend anymore!")
-
-        output = RedirectedTestOutput()  # Initialize each command
+
+def test_list_empty():
+    client = TestClient()
+    client.run("profile list")
+    assert "Profiles found in the cache:" in client.out
+
+
+def test_list():
+    client = TestClient()
+    profiles = ["default", "profile1", "profile2", "profile3",
+                "nested" + os.path.sep + "profile4",
+                "nested" + os.path.sep + "two" + os.path.sep + "profile5",
+                "nested" + os.path.sep + "profile6"]
+    if platform.system() != "Windows":
+        profiles.append("symlink_me" + os.path.sep + "profile7")
+    for profile in profiles:
+        save(os.path.join(client.cache.profiles_path, profile), "")
+
+    if platform.system() != "Windows":
+        os.symlink(os.path.join(client.cache.profiles_path, 'symlink_me'),
+                   os.path.join(client.cache.profiles_path, 'link'))
+        # profile7 will be shown twice because it is symlinked.
+        profiles.append("link" + os.path.sep + "profile7")
+
+    # Make sure local folder doesn't interact with profiles
+    os.mkdir(os.path.join(client.current_folder, "profile3"))
+    client.run("profile list")
+    for p in profiles:
+        assert p in client.out
+
+    # Test profile list json file
+    client.run("profile list --format=json", redirect_stdout="profiles_list.json")
+    json_content = client.load("profiles_list.json")
+    json_obj = json.loads(json_content)
+    assert type(json_obj) == list
+    assert set(profiles) == set(json_obj)
+
+
+def test_show():
+    client = TestClient()
+    profile1 = textwrap.dedent("""
+        [settings]
+        os=Windows
+        [options]
+        MyOption=32
+        [conf]
+        tools.build:jobs=20
+        """)
+
+    client.save({"profile1": profile1})
+    client.run("profile show -pr=profile1")
+    assert "[settings]\nos=Windows" in client.out
+    assert "MyOption=32" in client.out
+    assert "tools.build:jobs=20" in client.out
+
+    client.run("profile show -pr=profile1 -pr:b=profile1 -f=json", redirect_stdout="profile1.json")
+    profile = json.loads(client.load("profile1.json"))
+    assert profile == {'host': {'settings': {'os': 'Windows'}, 'package_settings': {},
```

Comment:
Addition to show that profile show properly works with its -f formatter, just in case